core/src/main/scala/kafka/zk/KafkaZkClient.scala

```diff
@@ -157,7 +157,7 @@ class KafkaZkClient private[zk] (zooKeeperClient: ZooKeeperClient,
     val namespaces = getChildren(FederatedTopicsZNode.path)
     namespaces
       // For all topics, generate (topic -> namespace) tuple
-      .flatMap(namespace => getAllFederatedTopicsInNamespace(namespace).map(_ -> namespace))
+      .flatMap(namespace => getAllFederatedTopicsInNamespace(namespace, registerWatch = true).map(_ -> namespace))
```

Comment:
This control is should be decided by kafka-server.  Let's expose this at L156 (with default `false`)